contributors/design-proposals/local-storage-overview.md

```diff
@@ -0,0 +1,604 @@
+3. The addon will monitor the health of secondary partitions and mark PVs as unhealthy whenever the backing local storage devices have failed.
+4. Bob creates a specialized controller (Operator) for his distributed filesystem and deploys it.
+5. The operator will identify all the nodes that it can schedule pods onto and discovers the PVs available on each of those nodes. The operator has a label selector that identifies the specific PVs that it can use (this helps preserve fast PVs for Databases for example).
+6. The operator will then create PVCs and manually bind to individual local PVs across all its nodes.
+7. It will then create pods, manually place them on specific nodes (similar to a DaemonSet) with high enough priority and have them use all the PVCs created by the Operator on those nodes.
+8. If a pod dies, it will get replaced with a new pod that uses the same set of PVCs that the old pod had used.
+9. If a PV gets marked as unhealthy, the Operator is expected to delete pods if they cannot tolerate device failures
+
+### Phippy manages a cluster and intends to mitigate storage I/O abuse
+
+1. Phippy creates a dedicated partition with a separate device for her system daemons. She achieves this by making `/var/log/containers`, `/var/lib/kubelet`, `/var/lib/docker` (with the docker runtime) all reside on a separate partition.
+2. Phippy is aware that pods can cause abuse to each other.
+3. Whenever a pod experiences I/O issues with it's EmptyDir volume, Phippy reconfigures those pods to use Persistent Volumes whose lifetime is tied to the pod.
+    ```yaml
+    apiVersion: v1
+    kind: pod
+    metadata:
+     name: foo
+    spec:
+     containers:
+     - name: fooc
+       resources:
+       limits:
+         storage.kubernetes.io/logs: 500Mi
+         storage.kubernetes.io/overlay: 1Gi
+       volumeMounts:
+       - name: myEphemeralPersistentVolume
+         mountPath: /mnt/tmpdata
+     volumes:
+     - name: myEphemeralPeristentVolume
+       inline:
+         spec:
+           accessModes: [ "ReadWriteOnce" ]
+           storageClassName: local-fast
+           resources:
+             limits:
+               size: 1Gi
+    ```
+
+4. Phippy notices some of her pods are experiencing spurious downtimes. With the help of monitoring (`iostat`), she notices that the nodes pods are running on are overloaded with I/O operations. She then updates her pods to use Logging Volumes which are backed by persistent storage. If a logging volumeMount is associated with a container, Kubelet will place log data from stdout & stderr of the container under the volume mount path within the container. Kubelet will continue to expose stdout/stderr log data to external logging agents using symlinks as it does already.
+
+    ```yaml
+    apiVersion: v1
+    kind: pod
+    metadata:
+     name: foo
+    spec:
+     containers:
+     - name: fooc
+       volumeMounts:
+       - name: myLoggingVolume
+         mountPath: /var/log/
+         policy:
```

Comment:
IIUC this "policy" field is new -- you might want to mention that explicitly (unless I'm mistaken about it being new), as I see you use it in a couple of the examples.